src/pages/Contacts.jsx

```diff
@@ -0,0 +1,160 @@
+// React Imports
+import React, { useState } from 'react';
+// Material UI Imports
+import AddIcon from '@mui/icons-material/Add';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Container from '@mui/material/Container';
+import Typography from '@mui/material/Typography';
+// Component Imports
+import { useDataset, useSession } from '@hooks';
+import { AddContactModal, DeleteContactModal } from '@components/Modals';
+import { ContactListTable } from '@components/Contacts';
+import { LoadingAnimation, EmptyListNotification } from '@components/Notification';
+import {
+  getStringNoLocale,
+  getThing,
+  removeThing,
+  getThingAll,
+  getUrl,
+  buildThing,
+  createThing,
+  setThing,
+  saveSolidDatasetAt
+} from '@inrupt/solid-client';
+import { RDF_PREDICATES } from '@constants';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+/**
+ * Contacts Component - Component that generates Contacts Page for PASS
+ *
+ * @memberof Pages
+ * @name Contacts
+ */
+
+const Contacts = () => {
+  localStorage.setItem('restorePath', '/clients');
+  // state for AddContactModal component
+  const { session, podUrl } = useSession();
+  const listUrl = new URL('PASS/Users/userlist.ttl', podUrl);
+  const [showAddContactModal, setShowAddContactModal] = useState(false);
+
+  const [showDeleteContactModal, setShowDeleteContactModal] = useState(false);
+  const [selectedContactToDelete, setSelectedContactToDelete] = useState(null);
+  const queryClient = useQueryClient();
+
+  const handleSelectDeleteContact = (contact) => {
+    setSelectedContactToDelete(contact);
+    setShowDeleteContactModal(true);
+  };
+
+  const parseContacts = (data) => {
```

Comment:
I've noticed some of the utility functions are being moved from other files and into the pages and component file.

Should we not try to separate the two and mainly keep handler functions/hooks within components? Think if we leave the utility functions outside in a separate file like before, it'll be easier to isolate and unit test.

Reply:
Yeah the main reason I removed those is because React query took over a lot of the code that was being written in those files. React Query also uses a different paradigm of 'queries' and 'mutations' that I found difficult to map directly onto the 'model' paradigm I was originally going for.

Either way, you're right we can probably factor these out a bit more. However, I don't think it will look exactly like the old `userList` file

Reply:
Moved these all into a custom hook